primefaces/src/main/java/org/primefaces/component/export/DataExporters.java

```diff
@@ -43,15 +43,22 @@ private DataExporters() {
     }
 
     public static <T extends UIComponent> Exporter<T> get(Class<T> targetClass, String type) {
-        String newType = type.toLowerCase();
         PrimeApplicationContext primeAppContext = PrimeApplicationContext.getCurrentInstance(FacesContext.getCurrentInstance());
-        Map<String, Class<? extends Exporter<?>>> supportedExporters = Optional.ofNullable(primeAppContext.getExporters().get(targetClass))
+
+        // since users might defined their own impl of components (e.g MyDataTable, MyTreeTable etc.)
+        // retrieve the most eligible built-in component class first (e.g DataTable, TreeTable) to get corresponding exporters
+        Class<? extends UIComponent> targetClassTmp = primeAppContext.getExporters().keySet().stream()
```

Comment:
this looks great just minor nitpick would be `resolvedTargetClass` instead of `targetClassTmp` ?

Reply:
Thank you! Sounds very good, will make the change